core/server/models/offer.js

```diff
@@ -0,0 +1,53 @@
+const ghostBookshelf = require('./base');
+
+const Offer = ghostBookshelf.Model.extend({
+    tableName: 'offers',
+
+    async onSaving(model, _attr, options) {
+        ghostBookshelf.Model.prototype.onSaving.apply(this, arguments);
+
+        if (model.get('name')) {
+            model.set('name', model.get('name').trim());
+        }
+
+        if (model.hasChanged('slug') || !model.get('slug')) {
+            const slug = model.get('slug') || model.get('name');
+
+            if (slug) {
+                const cleanSlug = await ghostBookshelf.Model.generateSlug(Offer, slug, {
+                    transacting: options.transacting
+                });
+
+                model.set({slug: cleanSlug});
+            }
+        }
+    },
+
+    /**
+     * The base model keeps only the columns, which are defined in the schema.
+     * We have to add the relations on top, otherwise bookshelf-relations
+     * has no access to the nested relations, which should be updated.
+     */
+    permittedAttributes: function permittedAttributes() {
+        let filteredKeys = ghostBookshelf.Model.prototype.permittedAttributes.apply(this, arguments);
+
+        this.relationships.forEach((key) => {
+            filteredKeys.push(key);
+        });
+
+        return filteredKeys;
+    },
+
+    product() {
+        return this.belongsTo('Product', 'product_id', 'id');
+    }
+});
+
+const Offers = ghostBookshelf.Collection.extend({
```

Comment:
Can remove this line - we're not adding collection going forward!

Reply:
Ah ok cool, updating 👍

Reply:
Done